presto-main/src/main/java/com/facebook/presto/operator/scalar/JsonOperators.java

```diff
@@ -224,6 +226,44 @@ public static Double castToDouble(@SqlType(JSON) Slice json)
 
     @ScalarOperator(CAST)
     @SqlNullable
+    @SqlType(REAL)
+    public static Long castToFloat(@SqlType(JSON) Slice json)
+    {
+        try (JsonParser parser = createJsonParser(JSON_FACTORY, json)) {
+            parser.nextToken();
+            Long result;
+            switch (parser.getCurrentToken()) {
+                case VALUE_NULL:
+                    result = null;
+                    break;
+                case VALUE_STRING:
+                    result = VarcharOperators.castToFloat(Slices.utf8Slice(parser.getText()));
+                    break;
+                case VALUE_NUMBER_FLOAT:
+                    result = (long) floatToRawIntBits(parser.getFloatValue());
+                    break;
+                case VALUE_NUMBER_INT:
+                    result = (long) floatToRawIntBits(parser.getFloatValue());
```

Comment:
Add comment here for justification (just like we did in `castToDouble`).

Even though I wrote that code (in `castToDouble` originally), when I saw this line, it's not obvious at all to me why we are doing this (instead of `BigintOperators.castToFloat`).